hw/dv/dpi/usbdpi/usbdpi.c

```diff
@@ -167,6 +168,24 @@ void usbdpi_device_to_host(void *ctx_void, const svBitVecVal *usb_d2p) {
     ssize_t written = fwrite(obuf, sizeof(char), (size_t)n, ctx->mon_file);
     assert(written == n);
   }
+
+  // Device-to-Host EOP
+  if (ctx->state == ST_GET && dp == 0 && dp == 0) {
```

Comment:
This whole file really needs a rewrite, especially the addition of proper decoding of the device-to-host path. However, detecting SE0 from the device at least lets us find a packet boundary, even if we don't know which packet that is. I figure we can leave doing a good DPI module for a later PR.

Reply:
Yes, sorry about this code. It was intended as a quick hack and grew out of control.

Reply:
Haha, no worries! It was nice to have some stimulus to generate waveforms, since the FPGA build times are longer, and attempts to work in an ILA met with failure.

I've only made it more messy with yet another state machine layered on top, but we'll clean it all up eventually. :)